src/measure_theory/function/conditional_expectation.lean

```diff
@@ -1814,37 +1814,52 @@ section condexp
 
 open_locale classical
 
-variables {𝕜} {m m0 : measurable_space α} {μ : measure α}
-  {hm : m ≤ m0} [sigma_finite (μ.trim hm)] {f g : α → F'} {s : set α}
+variables {𝕜} {m m0 : measurable_space α} {μ : measure α} {f g : α → F'} {s : set α}
 
 variables (m)
-/-- Conditional expectation of a function. Its value is 0 if the function is not integrable. -/
-@[irreducible] def condexp (hm : m ≤ m0) (μ : measure α) [sigma_finite (μ.trim hm)] (f : α → F') :
+/-- Conditional expectation of a function. Its value is 0 if the function is not integrable, if
+the σ-algebra is not a sub-σ-algebra or if the measure is not σ-finite on that σ-algebra. -/
+@[irreducible] def condexp (μ : measure α) (f : α → F') :
   α → F' :=
-if (strongly_measurable[m] f ∧ integrable f μ) then f
-else ae_strongly_measurable'_condexp_L1.mk (condexp_L1 hm μ f)
+if hm : m ≤ m0
+  then if hμ : sigma_finite (μ.trim hm)
+    then if (strongly_measurable[m] f ∧ integrable f μ)
+      then f
+      else (@ae_strongly_measurable'_condexp_L1 _ _ _ _ _ m m0 μ hm hμ _).mk
+        (@condexp_L1 _ _ _ _ _ _ _ hm μ hμ f)
+    else 0
+  else 0
 
 variables {m}
 
--- We define notations `μ[f|hm]` and `μ[f|m,hm]` for the conditional expectation of `f` with
--- respect to `m`. Both can be used in code but only the second one will be used by the goal view.
--- The first notation avoids the repetition of `m`, which is already present in `hm`. The second
--- one ensures that `m` stays visible in the goal view: when `hm` is complicated, it gets rendered
--- as `_` and the measurable space would not be visible in `μ[f|_]`, but is clear in `μ[f|m,_]`.
-localized "notation  μ `[` f `|` hm `]` := measure_theory.condexp _ hm μ f" in measure_theory
-localized "notation  μ `[` f `|` m `,` hm `]` := measure_theory.condexp m hm μ f" in measure_theory
+-- We define notation `μ[f|m]` for the conditional expectation of `f` with respect to `m`.
+localized "notation  μ `[` f `|` m `]` := measure_theory.condexp m μ f" in measure_theory
+
+lemma condexp_of_not_le (hm_not : ¬ m ≤ m0) : μ[f|m] = 0 := by rw [condexp, dif_neg hm_not]
+
+lemma condexp_of_not_sigma_finite (hm : m ≤ m0) (hμm_not : ¬ sigma_finite (μ.trim hm)) :
+  μ[f|m] = 0 :=
+by rw [condexp, dif_pos hm, dif_neg hμm_not]
 
-lemma condexp_of_strongly_measurable
+lemma condexp_of_sigma_finite (hm : m ≤ m0) (hμm : sigma_finite (μ.trim hm)) :
```

Comment:
Why is `hμm` a regular variable here, and a typeclass variable in the next lemma?